src/coreclr/vm/debugdebugger.cpp

```diff
@@ -532,182 +535,208 @@ FCIMPL4(void, DebugStackTrace::GetStackFramesInternal,
                 }
             }
 #endif
-            BOOL fPortablePDB = TRUE;
-
             // Check if the user wants the filenumber, linenumber info and that it is possible.
             if (!fIsEnc && fNeedFileInfo)
             {
 #ifdef FEATURE_ISYM_READER
-                BOOL fFileInfoSet = FALSE;
-                ULONG32 sourceLine = 0;
-                ULONG32 sourceColumn = 0;
-                WCHAR wszFileName[MAX_LONGPATH];
-                ULONG32 fileNameLength = 0;
+                BOOL fPortablePDB = FALSE;
+                if (pModule->GetPEAssembly()->HasLoadedPEImage())
                 {
-                    // Note: we need to enable preemptive GC when accessing the unmanages symbol store.
-                    GCX_PREEMP();
-
-                    // Note: we use the NoThrow version of GetISymUnmanagedReader. If getting the unmanaged
-                    // reader fails, then just leave the pointer NULL and leave any symbol info off of the
-                    // stack trace.
-                    ReleaseHolder<ISymUnmanagedReader> pISymUnmanagedReader(
-                        pModule->GetISymUnmanagedReaderNoThrow());
-
-                    if (pISymUnmanagedReader != NULL)
+                    PEDecoder* pe = pModule->GetPEAssembly()->GetLoadedLayout();
+                    IMAGE_DATA_DIRECTORY* debugDirectoryEntry = pe->GetDirectoryEntry(IMAGE_DIRECTORY_ENTRY_DEBUG);
+                    if (debugDirectoryEntry != nullptr)
+                    {
+                        IMAGE_DEBUG_DIRECTORY* debugDirectory = (IMAGE_DEBUG_DIRECTORY*)pe->GetDirectoryData(debugDirectoryEntry);
+                        if (debugDirectory != nullptr)
+                        {
+                            size_t nbytes = 0;
+                            while (nbytes < debugDirectoryEntry->Size)
+                            {
+                                if ((debugDirectory->Type == IMAGE_DEBUG_TYPE_CODEVIEW && debugDirectory->MinorVersion == PORTABLE_PDB_MINOR_VERSION) ||
+                                    (debugDirectory->Type == IMAGE_DEBUG_TYPE_EMBEDDED_PORTABLE_PDB))
+                                {
+                                    fPortablePDB = TRUE;
+                                    break;
+                                }
+                                debugDirectory++;
+                                nbytes += sizeof(*debugDirectory);
+                            }
+                        }
+                    }
+                }
+                if (!fPortablePDB)
+                {
+                    fPortablePDB = TRUE;
```

Comment:
Do we expect this flag to ever remain true given that we already checked for portable PDB above explicitly? A comment would help to say if this is just defensive coding or if there is a known case that the logic above isn't going to detect.